### PR TITLE
Move SPDX utilities to their own module

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1140)

### CODE REVIEW

File: spdx-utils/src/main/kotlin/SpdxUtils.kt [hunk 1]
```diff
@@ -67,6 +63,12 @@ fun calculatePackageVerificationCode(files: List<File>) =
             file.inputStream().use { DigestUtils.sha1Hex(it) }
         })
 
+/**
+ * A Kotlin-style convenience function to replace EnumSet.of() and EnumSet.noneOf().
+ */
+inline fun <reified T : Enum<T>> enumSetOf(vararg elems: T): EnumSet<T> =
+        EnumSet.noneOf(T::class.java).apply { addAll(elems) }
+
```
Comment (mnonnenmacher) [inline spdx-utils/src/main/kotlin/SpdxUtils.kt]: That function should be next to `EnumSet.plus` in `Extensions.kt`.

File: spdx-utils/src/main/kotlin/SpdxUtils.kt [hunk 2]
```diff
@@ -78,15 +80,8 @@ fun getLicenseText(id: String, handleExceptions: Boolean = false): String {
...
 
-        if (response.code() != HttpURLConnection.HTTP_OK || body == null) {
-            throw IOException("Failed to download the text for license '$id' from $url.")
-        }
-
-        ObjectMapper().readTree(body.string())[key].textValue()
-    }
+    // TODO: Change this to use built-in resources instead of making HTTP requests, but for now only make it independent
+    // of "OkHttpClientHelper" from the "utils" module.
+    return ObjectMapper().readTree(URL(url).readText())[key].textValue()
```
Comment (mnonnenmacher) [inline spdx-utils/src/main/kotlin/SpdxUtils.kt]: That function should also be in `Extensions.kt`.

File: spdx-utils/src/main/kotlin/SpdxUtils.kt (hunk 1 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: Why? It's not an extension function after all.

File: spdx-utils/src/main/kotlin/SpdxUtils.kt (hunk 2 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxUtils.kt]: Again, why? This also is no extension function, just a temporary object creation.

**Pull request merged**